src/EnergyPlus/ThermalComfort.cc

```diff
@@ -2107,57 +2085,60 @@ namespace ThermalComfort {
         // Return value
         Real64 CalcSurfaceWeightedMRT = 0.0;
 
-        // Local variables
-        int SurfNum2;  // surface number used in "for" loop
-        int ZoneNum2;  // zone number index
-        Real64 SumAET; // Intermediate calculational variable (area*emissivity*T) sum
-
         // Initialize ZoneAESum for all zones and SurfaceAE for all surfaces at the start of the simulation
         if (state.dataThermalComforts->FirstTimeSurfaceWeightedFlag) {
             state.dataThermalComforts->FirstTimeError = true;
             state.dataThermalComforts->FirstTimeSurfaceWeightedFlag = false;
-            state.dataThermalComforts->SurfaceAE.allocate(state.dataSurface->TotSurfaces);
-            state.dataThermalComforts->ZoneAESum.allocate(state.dataGlobal->NumOfZones);
-            state.dataThermalComforts->SurfaceAE = 0.0;
-            state.dataThermalComforts->ZoneAESum = 0.0;
-            for (SurfNum2 = 1; SurfNum2 <= state.dataSurface->TotSurfaces; ++SurfNum2) {
-                if (state.dataSurface->Surface(SurfNum2).HeatTransSurf) {
-                    state.dataThermalComforts->SurfaceAE(SurfNum2) =
-                        state.dataSurface->Surface(SurfNum2).Area *
-                        state.dataConstruction->Construct(state.dataSurface->Surface(SurfNum2).Construction).InsideAbsorpThermal;
-                    ZoneNum2 = state.dataSurface->Surface(SurfNum2).Zone;
-                    // Do NOT include the contribution of the Surface that is being surface weighted in this calculation since it will already be
-                    // accounted for
-                    if ((ZoneNum2 > 0) && (SurfNum2 != SurfNum))
-                        state.dataThermalComforts->ZoneAESum(ZoneNum2) += state.dataThermalComforts->SurfaceAE(SurfNum2);
+            for (auto const &thisRadEnclosure : state.dataViewFactor->EnclRadInfo) {
+                for (int const SurfNum2 : thisRadEnclosure.SurfacePtr) {
```

Comment:
The heart of all of this - loop over enclosure surfaces instead of zone surfaces.